File: backend/app/schemas/reconstruction.py
from pydantic import BaseModel


class VertebraBoundsSchema(BaseModel):
    min_z: float
    max_z: float
    center: list[float]


class FractureMarkerSchema(BaseModel):
    vertebre: str
    position: list[float]
    score: float


class Reconstruction3DSchema(BaseModel):
    vertices: list[list[float]]
    faces: list[list[int]]
    normals: list[list[float]]
    vertex_colors: list[str]
    vertebrae_bounds: dict[str, VertebraBoundsSchema]
    fracture_markers: list[FractureMarkerSchema] = []

File: frontend/src/types/reconstruction.ts
export interface VertebraBounds {
  min_z: number
  max_z: number
  center: [number, number, number]
}

export interface FractureMarker {
  vertebre: string
  position: [number, number, number]
  score: number
}

export interface Reconstruction3D {
  vertices: number[][]
  faces: number[][]
  normals: number[][]
  vertex_colors: string[]
  vertebrae_bounds: Record<string, VertebraBounds>
  fracture_markers: FractureMarker[]
}
